met-web/src/components/common/Modals/FeedbackModal.tsx

```diff
@@ -0,0 +1,342 @@
+    const [isOpen, setIsOpen] = useState(false);
+    const [feedbackFormData, setFeedbackFormData] = useState(createDefaultFeedback());
+    const { comment, rating, commentType } = feedbackFormData;
+
+    const StyledRating = styled(Rating)(({ theme }) => ({
+        '& .MuiRating-iconEmpty .MuiSvgIcon-root': {
+            opacity: '0.5',
+        },
+        textAlign: 'center',
+    }));
+
+    const customIcons: {
+        [index: string]: {
+            icon: React.ReactElement;
+            label: string;
+        };
+    } = {
+        5: {
+            icon: <SvgIcon fontSize="large" component={VeryDissatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 1 }} />,
+            label: 'Very Dissatisfied',
+        },
+        4: {
+            icon: <SvgIcon fontSize="large" component={DissatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 2 }} />,
+            label: 'Dissatisfied',
+        },
+        3: {
+            icon: <SvgIcon fontSize="large" component={NeutralIcon} viewBox="0 0 64 64" sx={{ marginX: 1 }} />,
+            label: 'Neutral',
+        },
+        2: {
+            icon: <SvgIcon fontSize="large" component={SatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 2 }} />,
+            label: 'Satisfied',
+        },
+        1: {
+            icon: <SvgIcon fontSize="large" component={VerySatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 1 }} />,
+            label: 'Very Satisfied',
+        },
+    };
+
+    function IconContainer(props: IconContainerProps) {
+        const { value, ...other } = props;
+        return <span {...other}>{customIcons[value].icon}</span>;
+    }
+
+    function getLabelText() {
+        if (rating == 0) return;
+        return `${customIcons[rating].label}`;
+    }
+
+    function getPlaceholderText() {
+        switch (commentType) {
+            case 'Issue':
+                return 'Something does not work...';
+            case 'Idea':
+                return 'I would like to see...';
+            case 'Else':
+                return 'I was wondering...';
+        }
+
+        return '';
+    }
+
+    function handleRatingChanged(value: number) {
+        setFeedbackFormData({
+            ...feedbackFormData,
+            rating: rating == value ? 0 : value,
+        });
+    }
+
+    function handleCommentTypeChanged(value: string) {
+        setFeedbackFormData({
+            ...feedbackFormData,
+            commentType: commentType == value ? '' : value,
+        });
+    }
+
+    function handleCommentChanged(value: string) {
+        setFeedbackFormData({
+            ...feedbackFormData,
+            comment: value,
+        });
+    }
+
+    function handleSubmit() {
+        console.log(feedbackFormData);
+        setIsSubmitted(true);
+    }
+
+    function handleClose() {
+        setIsSubmitted(false);
+        setFeedbackFormData(createDefaultFeedback());
+        setIsOpen(false);
+    }
+
+    const StyledButton = styled(Button)(() => ({
```

Comment:
a more specific name for the StyledButton perhaps

Reply:
Renamed to CommentTypeButton